cumulus_etl/loaders/fhir/ndjson_loader.py

```diff
@@ -85,9 +88,21 @@ async def load_resources(self, resources: set[str]) -> base.LoaderResults:
         filenames = common.ls_resources(input_root, resources, warn_if_empty=True)
         for filename in filenames:
             input_root.get(filename, f"{tmpdir.name}/")
+            # Decompress any *.gz files, because the MS tool can't understand them
+            self._decompress_file(f"{tmpdir.name}/{os.path.basename(filename)}")
 
         return self.read_loader_results(input_root, tmpdir)
 
+    @staticmethod
+    def _decompress_file(path: str):
+        if not path.casefold().endswith(".gz"):
+            return
+        target_path = path[:-3]
+        with gzip.open(path, "rb") as f_in:
+            with open(target_path, "wb") as f_out:
+                shutil.copyfileobj(f_in, f_out)
+        os.remove(path)
```

Comment:
This is probably a worse structural overhead thing, but - rather than replacing a compressed file with the unzipped file, for disk space usage a pattern like 'for each file in files: unzip it, do all the processing, and then remove the unzipped one before continuing' might be nice?

Reply:
Ah fair, but this code flow is only for "copy the original file (maybe zipped) to a tmpdir, then process it" - so we already are working in a separate tmpdir and not the original input files. So no need to keep the original zip file in that case - we won't use it and it's not the user's actual copy of the file.

Reply:
I was a little shocked that there isn't like a `shutil` utility method to just decompress a file as a one-liner. I guess questions like this are why not, but :shrug: - I'm finding that the "how to deal with compressed files" part of Python is all very low level with almost no abstraction layers.